### DISCUSSION TITLE
Game server won't load anymore

### DISCUSSION THREAD

**Discussion by TheDakk** [mbround18/valheim-docker#218] [Help!]
Game server won't load anymore this is what showing in the logs, can't find anything useful:

valheim_1  | Connecting anonymously to Steam Public...Logged in OK
valheim_1  | Waiting for user info...OK
valheim_1  | Success! App '896660' already up to date.
valheim_1  | [ODIN][INFO]  - Successfully installed Valheim!
valheim_1  | [Valheim][steam]: Starting server...
valheim_1  | [ODIN][DEBUG] - Debugging set to true
valheim_1  | [ODIN][DEBUG] - Debug mode enabled!
valheim_1  | [ODIN][DEBUG] - Launching start command...
valheim_1  | [ODIN][INFO]  - Setting up start scripts...
valheim_1  | [ODIN][DEBUG] - Loading config file...
valheim_1  | [ODIN][DEBUG] - Config file set to: config.json
valheim_1  | [ODIN][DEBUG] - Checking password compliance...
valheim_1  | [ODIN][DEBUG] - Dry run condition: false
valheim_1  | [ODIN][INFO]  - Looking for burial mounds...
valheim_1  | [ODIN][DEBUG] - Using default env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Found env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Using default env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Found env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Using default env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Found env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Using default env var 'LD_PRELOAD': ''
valheim_1  | [ODIN][DEBUG] - Found env var 'LD_PRELOAD': ''
valheim_1  | [ODIN][DEBUG] - Using default env var 'DOORSTOP_LIB': 'libdoorstop_x64.so'
valheim_1  | [ODIN][DEBUG] - Found env var 'DOORSTOP_LIB': 'libdoorstop_x64.so'
valheim_1  | [ODIN][DEBUG] - Using default env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Found env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Using default env var 'DOORSTOP_LIBS': '/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Found env var 'DOORSTOP_LIBS': '/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Using default env var 'LD_LIBRARY_PATH': './linux64:/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Found env var 'LD_LIBRARY_PATH': './linux64:/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Using default env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Found env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Using default env var 'DOORSTOP_INVOKE_DLL_PATH': '/home/steam/valheim/BepInEx/core/BepInEx.Preloader.dll'
valheim_1  | [ODIN][DEBUG] - Found env var 'DOORSTOP_INVOKE_DLL_PATH': '/home/steam/valheim/BepInEx/core/BepInEx.Preloader.dll'
valheim_1  | [ODIN][DEBUG] - Using default env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Found env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Using default env var 'DOORSTOP_CORLIB_OVERRIDE_PATH': '/home/steam/valheim/unstripped_corlib'
valheim_1  | [ODIN][DEBUG] - Found env var 'DOORSTOP_CORLIB_OVERRIDE_PATH': '/home/steam/valheim/unstripped_corlib'
valheim_1  | [ODIN][DEBUG] - Using default env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Found env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Using default env var 'DOORSTOP_LIBS': '/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Found env var 'DOORSTOP_LIBS': '/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Using default env var 'DYLD_LIBRARY_PATH': '/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Found env var 'DYLD_LIBRARY_PATH': '/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Using default env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Found env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Using default env var 'DOORSTOP_LIBS': '/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Found env var 'DOORSTOP_LIBS': '/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Using default env var 'DOORSTOP_LIB': 'libdoorstop_x64.so'
valheim_1  | [ODIN][DEBUG] - Found env var 'DOORSTOP_LIB': 'libdoorstop_x64.so'
valheim_1  | [ODIN][DEBUG] - Using default env var 'DYLD_INSERT_LIBRARIES': '/home/steam/valheim/doorstop_libs/libdoorstop_x64.so'
valheim_1  | [ODIN][DEBUG] - Found env var 'DYLD_INSERT_LIBRARIES': '/home/steam/valheim/doorstop_libs/libdoorstop_x64.so'
valheim_1  | [ODIN][DEBUG] - Using default env var 'DYLD_INSERT_LIBRARIES': '/home/steam/valheim/doorstop_libs/libdoorstop_x64.so'
valheim_1  | [ODIN][DEBUG] - Found env var 'DYLD_INSERT_LIBRARIES': '/home/steam/valheim/doorstop_libs/libdoorstop_x64.so'
valheim_1  | [ODIN][INFO]  - Checking for BepInEx Environment...
valheim_1  | [ODIN][DEBUG] - LD_PRELOAD: libdoorstop_x64.so
valheim_1  | [ODIN][DEBUG] - LD_LIBRARY_PATH: ./linux64:/home/steam/valheim/doorstop_libs
valheim_1  | [ODIN][DEBUG] - DOORSTOP_ENABLE: TRUE
valheim_1  | [ODIN][DEBUG] - DOORSTOP_INVOKE_DLL_PATH: /home/steam/valheim/BepInEx/core/BepInEx.Preloader.dll
valheim_1  | [ODIN][DEBUG] - DOORSTOP_CORLIB_OVERRIDE_PATH: /home/steam/valheim/unstripped_corlib
valheim_1  | [ODIN][DEBUG] - DYLD_LIBRARY_PATH: /home/steam/valheim/doorstop_libs
valheim_1  | [ODIN][DEBUG] - DYLD_INSERT_LIBRARIES: /home/steam/valheim/doorstop_libs/libdoorstop_x64.so
valheim_1  | [ODIN][DEBUG] - Checking for BepInEx specific files...
valheim_1  | [ODIN][DEBUG] - Path /home/steam/valheim/unstripped_corlib does not exist
valheim_1  | [ODIN][DEBUG] - Uhh ohh!!! Looks like you are missing something.
valheim_1  | [ODIN][INFO]  - Server has been started and Daemonized. It should be online shortly!
valheim_1  | [ODIN][INFO]  - Keep an eye out for 'Game server connected' in the log!
valheim_1  | [ODIN][INFO]  - (this indicates its online without any errors.)
valheim_1  | [Valheim][steam]: ###########################################################################
valheim_1  | [Valheim][steam]:
valheim_1  | Everything looks happy <3
valheim_1  |
valheim_1  | Check your output.log for 'Game server connected'
valheim_1  |
valheim_1  | [Valheim][steam]: ###########################################################################
valheim_1  | [Valheim][steam]: Initializing Variables....
valheim_1  | [ODIN][DEBUG] - Debugging set to true
valheim_1  | [ODIN][DEBUG] - Debug mode enabled!
valheim_1  | [ODIN][DEBUG] - Launching configure command...
valheim_1  | [ODIN][DEBUG] - Pulling config file...
valheim_1  | [ODIN][DEBUG] - Config file set to: config.json
valheim_1  | [ODIN][DEBUG] - Writing config file...
valheim_1  | [ODIN][DEBUG] - Using default env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Found env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Checking env for server_executable
valheim_1  | [ODIN][DEBUG] - Checking env for port
valheim_1  | [ODIN][DEBUG] - Env variable found port=2456
valheim_1  | [ODIN][DEBUG] - Checking env for name
valheim_1  | [ODIN][DEBUG] - Env variable found name=********
valheim_1  | [ODIN][DEBUG] - Checking env for world
valheim_1  | [ODIN][DEBUG] - Env variable found world=********
valheim_1  | [ODIN][DEBUG] - Checking env for public
valheim_1  | [ODIN][DEBUG] - Env variable found public=1
valheim_1  | [ODIN][DEBUG] - Checking env for password
valheim_1  | [ODIN][DEBUG] - Env variable found password=********
valheim_1  | [ODIN][DEBUG] - Writing config content:
valheim_1  | [ODIN][DEBUG] - {"port":"2456","name":"********","world":"********","public":"1","password":"*******","command":"/home/steam/valheim/valheim_server.x86_64"}
valheim_1  | [ODIN][INFO]  - Successfully written config.json
valheim_1  | [Valheim][steam]: Starting server...
valheim_1  | [ODIN][DEBUG] - Debugging set to true
valheim_1  | [ODIN][DEBUG] - Debug mode enabled!
valheim_1  | [ODIN][DEBUG] - Launching start command...
valheim_1  | [ODIN][INFO]  - Setting up start scripts...
valheim_1  | [ODIN][DEBUG] - Loading config file...
valheim_1  | [ODIN][DEBUG] - Config file set to: config.json
valheim_1  | [ODIN][DEBUG] - Checking password compliance...
valheim_1  | [ODIN][DEBUG] - Dry run condition: false
valheim_1  | [ODIN][INFO]  - Looking for burial mounds...
valheim_1  | [ODIN][DEBUG] - Using default env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Found env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Using default env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Found env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Using default env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Found env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Using default env var 'LD_PRELOAD': ''
valheim_1  | [ODIN][DEBUG] - Found env var 'LD_PRELOAD': ''
valheim_1  | [ODIN][DEBUG] - Using default env var 'DOORSTOP_LIB': 'libdoorstop_x64.so'
valheim_1  | [ODIN][DEBUG] - Found env var 'DOORSTOP_LIB': 'libdoorstop_x64.so'
valheim_1  | [ODIN][DEBUG] - Using default env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Found env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Using default env var 'DOORSTOP_LIBS': '/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Found env var 'DOORSTOP_LIBS': '/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Using default env var 'LD_LIBRARY_PATH': './linux64:/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Found env var 'LD_LIBRARY_PATH': './linux64:/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Using default env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Found env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Using default env var 'DOORSTOP_INVOKE_DLL_PATH': '/home/steam/valheim/BepInEx/core/BepInEx.Preloader.dll'
valheim_1  | [ODIN][DEBUG] - Found env var 'DOORSTOP_INVOKE_DLL_PATH': '/home/steam/valheim/BepInEx/core/BepInEx.Preloader.dll'
valheim_1  | [ODIN][DEBUG] - Using default env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Found env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Using default env var 'DOORSTOP_CORLIB_OVERRIDE_PATH': '/home/steam/valheim/unstripped_corlib'
valheim_1  | [ODIN][DEBUG] - Found env var 'DOORSTOP_CORLIB_OVERRIDE_PATH': '/home/steam/valheim/unstripped_corlib'
valheim_1  | [ODIN][DEBUG] - Using default env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Found env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Using default env var 'DOORSTOP_LIBS': '/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Found env var 'DOORSTOP_LIBS': '/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Using default env var 'DYLD_LIBRARY_PATH': '/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Found env var 'DYLD_LIBRARY_PATH': '/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Using default env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Found env var 'ODIN_WORKING_DIR': '/home/steam/valheim'
valheim_1  | [ODIN][DEBUG] - Using default env var 'DOORSTOP_LIBS': '/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Found env var 'DOORSTOP_LIBS': '/home/steam/valheim/doorstop_libs'
valheim_1  | [ODIN][DEBUG] - Using default env var 'DOORSTOP_LIB': 'libdoorstop_x64.so'
valheim_1  | [ODIN][DEBUG] - Found env var 'DOORSTOP_LIB': 'libdoorstop_x64.so'
valheim_1  | [ODIN][DEBUG] - Using default env var 'DYLD_INSERT_LIBRARIES': '/home/steam/valheim/doorstop_libs/libdoorstop_x64.so'
valheim_1  | [ODIN][DEBUG] - Found env var 'DYLD_INSERT_LIBRARIES': '/home/steam/valheim/doorstop_libs/libdoorstop_x64.so'
valheim_1  | [ODIN][DEBUG] - Using default env var 'DYLD_INSERT_LIBRARIES': '/home/steam/valheim/doorstop_libs/libdoorstop_x64.so'
valheim_1  | [ODIN][DEBUG] - Found env var 'DYLD_INSERT_LIBRARIES': '/home/steam/valheim/doorstop_libs/libdoorstop_x64.so'
valheim_1  | [ODIN][INFO]  - Checking for BepInEx Environment...
valheim_1  | [ODIN][DEBUG] - LD_PRELOAD: libdoorstop_x64.so
valheim_1  | [ODIN][DEBUG] - LD_LIBRARY_PATH: ./linux64:/home/steam/valheim/doorstop_libs
valheim_1  | [ODIN][DEBUG] - DOORSTOP_ENABLE: TRUE
valheim_1  | [ODIN][DEBUG] - DOORSTOP_INVOKE_DLL_PATH: /home/steam/valheim/BepInEx/core/BepInEx.Preloader.dll
valheim_1  | [ODIN][DEBUG] - DOORSTOP_CORLIB_OVERRIDE_PATH: /home/steam/valheim/unstripped_corlib
valheim_1  | [ODIN][DEBUG] - DYLD_LIBRARY_PATH: /home/steam/valheim/doorstop_libs
valheim_1  | [ODIN][DEBUG] - DYLD_INSERT_LIBRARIES: /home/steam/valheim/doorstop_libs/libdoorstop_x64.so
valheim_1  | [ODIN][DEBUG] - Checking for BepInEx specific files...
valheim_1  | [ODIN][DEBUG] - Path /home/steam/valheim/unstripped_corlib does not exist
valheim_1  | [ODIN][DEBUG] - Uhh ohh!!! Looks like you are missing something.
valheim_1  | [ODIN][INFO]  - Server has been started and Daemonized. It should be online shortly!
valheim_1  | [ODIN][INFO]  - Keep an eye out for 'Game server connected' in the log!
valheim_1  | [ODIN][INFO]  - (this indicates its online without any errors.)
valheim_1  | [Valheim][steam]: Herding Graydwarfs...
valheim_1  | ==> /home/steam/valheim/logs/valheim_server.log <==
valheim_1  |     Version:  NULL 1.0 [1.0]
valheim_1  |     Renderer: Null Device
valheim_1  |     Vendor:   Unity Technologies
valheim_1  | Begin MonoManager ReloadAssembly
valheim_1  | - Completed reload, in  0.108 seconds
valheim_1  | UnloadTime: 1.777406 ms
valheim_1  | 02/27/2021 12:47:42: Starting to load scene:start
valheim_1  |
valheim_1  | (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 35)
valheim_1  |
valheim_1  |
valheim_1  | ==> /home/steam/valheim/logs/valheim_server.err <==
valheim_1  |
valheim_1  | ==> /home/steam/valheim/logs/output.log <==
valheim_1  | [Valheim][steam]: ###########################################################################
valheim_1  | [Valheim][steam]: Valheim Server - Fri Feb 26 21:42:33 EST 2021
valheim_1  | [Valheim][steam]: STEAM_UID 1000 - STEAM_GUID 998
valheim_1  | [Valheim][steam]:
valheim_1  | Valheim Server Started...
valheim_1  |
valheim_1  | Keep an eye out for 'Game server connected' in the log!
valheim_1  | (this indicates its online without any errors.)
valheim_1  |
valheim_1  | [Valheim][steam]: ###########################################################################
valheim_1  |
valheim_1  | ==> /home/steam/valheim/logs/auto-update.out <==
valheim_1  |
valheim_1  | ==> /home/steam/valheim/logs/auto-backup.out <==

**Comment by mbround18** [comment]
Hey, looks like it should have started but the log got clipped. Can you drag and drop the 

```
/home/steam/Valheim/logs/valheim_server.log
```

And the 

```
/home/steam/Valheim/logs/valheim_server.err
```

Here I got the discussion?

**Reply by TheDakk** [reply]
[valheim_server.log](https://github.com/mbround18/valheim-docker/files/6055040/valheim_server.log)

It won't support the error log but it's empty.

**Reply by mbround18** [reply]
Huh lol not much different, can you try nuking your server installation and running `odin install` in the Valheim directory?

**Reply by TheDakk** [reply]
How do I "nuke" the installation?  just delete the "server" directory?

**Reply by TheDakk** [reply]
Also, I don't have Odin installed, I had trouble with it before.

**Reply by mbround18** [reply]
```
docker-compose exec --user steam valheim bash
```


Then inside the container

```
cd /home/steam/valheim
rm -rf ./*

odin install
```

Than exit the container and restart it

**Reply by TheDakk** [reply]
So I attempted to restart the docker container and all of sudden it started loading, no changes! Then I shut it down again and it didn't load all the way when I used the --detach command. Shut it down, then started it up again with just "docker-compose up" and now it loaded? I'm at a loss.. lol. I definitely don't think it's related to --detach command. It's been running fine for weeks!

**Reply by mbround18** [reply]
Lol try a docker-compose down, docker-compose pull, docker-compose up


See if that fixes things? If --detach is giving your grief that sounds like a funky docker thing 0.o

**Reply by TheDakk** [reply]
I actually ran a "pull" as part of my troubleshooting. I'm going to monitor and see if it hangs on auto update process tomorrow morning.

**Reply by TheDakk** [reply]
I wonder if this has to do with editing the assembly_valheim.dll to increase the bandwidth and prevent multiplayer lag.

**Reply by mbround18** [reply]
potentially, if you did it wrong or compiled it for the wrong system, than it wouldnt load. My suggestion is to use Longship which can handle these bitrate switches. (I think also valheim plus added this feature too. )

**Reply by TheDakk** [reply]
Do you have a link on how to use it? I’ve been using dnSpy 64-bit.

Found this post on Reddit which is very similar to what I’m seeing:

https://www.reddit.com/r/valheim/comments/lp9lmt/server_fix_for_multiplayer_lagdesync_dummy_proof/gofbnud/

**Reply by mbround18** [reply]
A few others who use this docker image are using the link below to help with lag:

https://github.com/AlexMog/Longship

**Reply by TheDakk** [reply]
So I ended up nuking everything and all, but still had issues. Turns it out it looks like it was not enough RAM. I was using a 4GB droplet and it was only loading 50% of the time. Once I went to 8GB, it started loading 100% of the time and quicker.

**Comment by TheDakk** [comment]
Needed to add more RAM to resolve server not always loading.

**Reply by mbround18** [reply]
Ahh, yeah that will happen :P the games pretty chaunky by itself and with mods can become even more chungus in terms of being a ram eater.
